version/src/main/java/io/smallrye/common/version/VersionRange.java

```diff
@@ -0,0 +1,61 @@
+package io.smallrye.common.version;
+
+import static io.smallrye.common.version.Messages.msg;
+
+import java.util.function.Predicate;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * A {@link VersionRange} is a predicate that tests if a version string is within a specified range.
+ */
+public class VersionRange implements Predicate<String> {
+
+    /**
+     * Range in format "[1.0,2.0)"
```

Comment:
We could probably borrow a more comprehensive version range impl from, for example, `org.apache.maven.artifact.versioning.VersionRange`?

Reply:
Ah that's a good idea, I am not checking for overlaps or caching the ranges, for example